src/js/components/Header/index.js

```diff
@@ -1,11 +1,51 @@
-import React from 'react';
-import ReactDOM from 'react-dom';
+// @flow
+import React, { Component } from 'react';
+import Logo from '../Logo';
+import Search from '../Search';
+import Poster from '../Poster';
+import SearchButton from '../SearchButton';
+import { views } from '../../helpers';
 
-const header =  React.createElement('div', {className: 'header'},
-  React.createElement('div', {className: 'title header__title'}, 'netflixroulette')
-);
+type Props = {
+  onSearch: (data: string, filter:string) => void,
+  toSearch: () => void,
+  view: string,
+  posterData: {
+    title: string,
+  },
+};
 
-ReactDOM.render(
-  header,
-  document.getElementById('header')
-);
+function topHeader(props: Props) {
+  const { view, toSearch } = props;
+  if ( view === views.POSTER ) {
+    return (
+      <div className='header__top'>
+        <Logo />
+        <SearchButton handleSearch={toSearch} />
+      </div>
+    );
+  }
+  return <div className='header__top'> <Logo /> </div>;
+}
+
+function contentHeader( props: Props ) {
```

Comment:
We don't need to have static functions and pass props around when we have classes. I would suggest moving these to class methods:
```jsx
class Header extends Component<Props> {
    renderTopHeader = () => {
        const { view, toSearch } = this.props;
         ....
    };
```

Reply:
change to a functional component